xcm/xcm-executor/src/lib.rs

```diff
@@ -46,8 +46,8 @@ pub struct FeesMode {
 
 const RECURSION_LIMIT: u8 = 10;
 
-thread_local! {
-	static RECURSION_COUNT: Cell<u8> = Cell::new(0);
+environmental::thread_local_impl! {
```

Comment:
```suggestion
environmental::environmental! {
```

Never ever use private types!!